app/script/cryptography/CryptographyMapper.js

```diff
@@ -0,0 +1,372 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.cryptography = z.cryptography || {};
+
+z.cryptography.CryptographyMapper = class CryptographyMapper {
+  constructor() {
+    this.logger = new z.util.Logger('z.cryptography.CryptographyMapper', z.config.LOGGER.OPTIONS);
+  }
+
+  /*
+  Maps a generic message into an event in JSON.
+
+  @param {z.proto.GenericMessage} generic_message Received ProtoBuffer message
+  @param {z.event.Backend.CONVERSATION.OTR-ASSET-ADD, z.event.Backend.CONVERSATION.OTR-MESSAGE-ADD} event Event
+  @return {Promise} - Resolves with the mapped event
+  */
+  map_generic_message(generic_message, event) {
+    return Promise.resolve()
+    .then(() => {
+      if (generic_message) {
+        return this._map_generic_message(generic_message, event);
+      }
+      throw new z.cryptography.CryptographyError(z.cryptography.CryptographyError.TYPE.NO_GENERIC_MESSAGE);
+    })
+    .then((specific_content) => {
+      return $.extend({
+        conversation: event.conversation,
+        id: generic_message.message_id,
+        from: event.from,
+        time: event.time,
+        status: event.status,
+      }
+      , specific_content);
+    });
```

Comment:
Maybe like this (because the ` , specific_content);` looks strange:

```js
.then((specific_content) => {
  return $.extend(
    {
      conversation: event.conversation,
      id: generic_message.message_id,
      from: event.from,
      time: event.time,
      status: event.status,
    },
    specific_content,
  );
});
```